Add ud resolution

V-Shadbolt opened the pull request:
## Explanation

Implements Unstoppable Domains Resolution Library
- Resolves UD on send and swap flow
- Resolves UD IPFS websites in address bar

## More Information

Responds to issue #12990

## Manual Testing Steps

[External MetaMask + UD Resolution Testing.pdf](https://github.com/MetaMask/metamask-extension/files/9559866/External.MetaMask.%2B.UD.Resolution.Testing.pdf)

### Testing Address Resolution

1. From the main screen press ‘Send’
2. Type Unstoppable Domain into search bar
3. Check to see if it resolves
4. It should resolve correctly for all UD TLDs
- .crypto
- .blockchain
- .bitcoin
- .zil
- .coin
- .nft 
- .wallet
- .888
- .dao
- .x

It should resolve for different currencies and tokens like MATIC & USDT

It should throw an error if
- The UD is not registered / is not valid OR if there is no address associated with the selected currency 
- The UD does not have an address associate with the selected currency / token
- After selecting a UD and the user swaps to a token that the UD doesn’t have an address for 

It should throw a warning if
- The UD contains a confusing character

### Testing IPFS Website Resolution
1. Simply type the UD into your browser’s address bar with a ‘/’ at the end 
2. On Chrome, Brave or other chrome based browsers it should
- Resolve all UD TLDs EXCEPT .888
- Redirect to IPFS website if a site exists
- Redirect to UD search page if IPFS website does not exist
3. On Firefox it should
- Resolve all UD TLDs
- Redirect to UD search page if IPFS website does not exist

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

V-Shadbolt committed:
Added UD resolution dependency

V-Shadbolt committed:
Add UD to placeholder text and sender/recipient js

V-Shadbolt committed:
Add UD Controller

V-Shadbolt committed:
Fix UD controller references and function names

V-Shadbolt committed:
Update policy files, controllers, and send pages

V-Shadbolt committed:
Log UNS resolution to Console

only support ETH address. Lots of errors.

V-Shadbolt committed:
Add UD Resolution error handling

Handling in-function for now

V-Shadbolt committed:
Added state to UNS resolution

V-Shadbolt committed:
Fixed ENS and UNS crossfiring

bbehrman10 committed:
adds TLD constant, UNS domain validation, & updates the input logic to avoid ENS & UNS crossfire

V-Shadbolt committed:
Add error handling for UNS resolution

bbehrman10 committed:
modularizes the UNS ducks implementation

V-Shadbolt committed:
Fixed Lint Spaced Error in Controller Build

V-Shadbolt committed:
Rename files and add missing UNS references

Changed ens-input to domain-input, uns to tld, and added initial UNS references to add contact flow

bbehrman10 committed:
single and multi chain resolution implemented

V-Shadbolt committed:
Fix merge conflicts with constants file

V-Shadbolt committed:
Fixed build issues from develop merge

Updated policy files

bbehrman10 committed:
adds resolving of UNS IPFS websites

V-Shadbolt committed:
Removed UD controller and references

V-Shadbolt committed:
Updated error handling to include multichain codes

bbehrman10 committed:
adds token swap mvp

V-Shadbolt committed:
added swap token error handling

V-Shadbolt committed:
Extended swap token error handling

bbehrman10 committed:
squashes swap token resolve bug

V-Shadbolt committed:
Removed console log

bbehrman10 committed:
converts static TLD and Currencies to APIs

V-Shadbolt committed:
Fixing develop merge conflicts

bbehrman10 committed:
removes console logs and some comments

V-Shadbolt committed:
Code cleanup

bbehrman10 committed:
adjusts IPFS resolution to use API, resolves timeout issue with IPFS sites

bbehrman10 committed:
updates ENS detection function, resolves floating error, ready for internal testing

bbehrman10 committed:
remove unnecessary file

bbehrman10 committed:
refactored code from feedback, lightened the API calls, minor UI updates to include UNS

V-Shadbolt committed:
Fixed merge conflict with UNS and ENS IPFS

bbehrman10 commented:
I have read the CLA Document and I hereby sign the CLA

brad-decker commented:
Hey @bbehrman10 and @V-Shadbolt -- thanks for the submission. Looks like a lot of test failures are valid. I'm going to change this to draft until you have tests passing and ready for review. Feel free to pull it out of draft when ready

bbehrman10 committed:
fixes most lint errors

bbehrman10 committed:
resolves more lint errors

bbehrman10 committed:
resolves state.UNS error and a spec error

bbehrman10 committed:
reverts a state issue

bbehrman10 committed:
working through unit tests

bbehrman10 committed:
matchObject instead of StrictEqual

bbehrman10 commented:
<img width="1068" alt="Screen Shot 2022-09-14 at 8 58 33 PM" src="https://user-images.githubusercontent.com/17752402/190303444-117c3001-0a84-4b40-8479-7dc2151d29db.png">
<img width="991" alt="Screen Shot 2022-09-14 at 8 58 40 PM" src="https://user-images.githubusercontent.com/17752402/190303447-146e22d9-eaa1-4e99-aa69-dc14ce8057fb.png">

bbehrman10 commented:
@brad-decker any thoughts on the two screenshots above? Not sure why the e2e tests are timing out, we didn't touch any of that stuff

bbehrman10 committed:
fixes bug with standard addresses

V-Shadbolt committed:
Removed try-catch in send.js

V-Shadbolt committed:
Replaces try catch in send.js

bbehrman10 committed:
fixes lint errors

bbehrman10 committed:
wraps api calls in try catch blocks

brad-decker commented on app/scripts/metamask-controller.js:
```diff
@@ -1743,6 +1743,7 @@ export default class MetamaskController extends EventEmitter {
         ),
       setFirstTimeUsedNetwork:
         appStateController.setFirstTimeUsedNetwork.bind(appStateController),
+
```
This change seems unccessary

brad-decker commented on ui/ducks/send/send.js:
```diff
@@ -1354,14 +1358,25 @@ const slice = createSlice({
           } = action.payload;
-
+          let validUDDomain = false;
+          try {
+            udTlds.payload.forEach((tld) => {
+              if (state.recipientInput.toLowerCase().endsWith(`.${tld}`)) {
+                validUDDomain = true;
+              }
+            });
+          } catch {
```
I don't believe this code will throw an error? perhaps if state.recipientInput is undefined but in no other cases that I can think of? Maybe just a optional chaining call if its possible that state.recipientInput may be undefined in this case and drop the try/catch?

bbehrman10 replied:
we were getting an error on the e2e tests and then the try catches seemed to resolve them, but we can investigate

brad-decker commented on ui/ducks/send/send.js:
```diff
@@ -1354,14 +1358,25 @@ const slice = createSlice({
               mixedCaseUseChecksum: true,
             }) &&
-              !isValidDomainName(state.recipientInput))
+              !isValidENSDomainName(state.recipientInput) &&
+              !validUDDomain)
```
For legibility reasons could we add a new intermediate variable before this if block: `isValidDomain` which checks if it is either a valid ENS or a valid UNS domain, then we can simplify this check to isValidDomain === false

brad-decker commented on ui/ducks/send/send.js:
```diff
@@ -1908,20 +1923,35 @@ export function updateRecipientUserInput(userInput) {
+            useTokenDetection,
+            tokenAddressList,
+            isProbablyAnAssetContract,
+            udTlds: state.UNS.tlds,
```
I assume this try/catch is because state.UNS.tlds could be undefined? this can be simplified to:

state.UNS?.tlds, which will then just pass 'udTlds' to undefined and avoids the try/catch.

brad-decker commented on ui/ducks/send/send.js:
```diff
@@ -2011,6 +2041,53 @@ export function updateSendAsset(
+          state.UNS.domainName
+        ) {
+          let unsError = null;
+          const object = await swapToken(
+            state.UNS.domainName,
+            state.metamask.provider?.ticker ?? 'ETH',
+          );
```
object is not an acceptable variable name, this needs to be renamed to reflect what this is.

brad-decker replied:
In addition, swapToken method name needs to be renamed, as it too closely aligns with MetaMask swaps. My first thought reading this was that it came from the swaps code not the UNS code.

bbehrman10 replied:
absolutely agree, we will make those changes

brad-decker commented on ui/ducks/send/send.js:
```diff
@@ -2011,6 +2041,53 @@ export function updateSendAsset(
           } with symbol ${state.metamask.provider?.ticker ?? ETH}`,
         ),
       );
+      if (state.UNS) {
```
Can you please describe what this section of code is doing? From my perspective it appears as though it is changing the asset based on the domain name resolved by UNS, is that accurate? It also appears to be changing the recipient?

bbehrman10 replied:
when you swap a token after selecting an address we need to pull the correct crypto address associated with the Unstoppable Domain. Because you can have multiple crypto addresses associated with one of our Domains we need to pull the correct one depending on which token you swap to

brad-decker commented on ui/ducks/send/send.js:
```diff
@@ -2115,6 +2191,38 @@ export function updateSendAsset(
           );
         }
       }
+      if (state.UNS) {
+        if (
+          state.UNS.resolution ===
+            state.send.draftTransactions[state.send.currentTransactionUUID]
+              .recipient.address &&
+          state.UNS.domainName
+        ) {
+          const object = await swapToken(state.UNS.domainName, asset);
+          if (object.error) {
+            if (
+              object.error === 'UnspecifiedCurrency' ||
+              object.error === 'RecordNotFound'
+            ) {
+              asset.error = UNS_CURRENCY_SPEC_ERROR;
+            } else if (object.error === 'UnsupportedCurrency') {
+              asset.error = UNS_CURRENCY_ERROR;
+            } else {
+              asset.error = UNS_UNKNOWN_ERROR;
+            }
+            await dispatch(actions.updateAsset({ asset, initialAssetSet }));
+            throw new Error('No address associated with this token');
+          } else {
+            dispatch(unsLookup(object));
+            await dispatch(
+              updateRecipient({
+                address: object.address,
+                nickname: object.unsName,
+              }),
+            );
+          }
+        }
+      }
```
This code is duplicated can you help me understand what this portion of the code is doing?

bbehrman10 replied:
we had this twice because when you swap a token it can either be a native token or a multi chain token

brad-decker commented on ui/helpers/utils/util.js:
```diff
@@ -87,17 +86,77 @@ export function addressSummary(
+  });
+  return result;
+}
+export async function getUdTlds() {
+  let result = [];
+  try {
+    const url = 'https://resolve.unstoppabledomains.com/supported_tlds';
+    const response = await fetch(url);
+    const data = await response.json();
+    result = data.tlds;
+    return result;
+  } catch {
+    return result;
+  }
+}
+
+export async function getAndParseUdCurrencies() {
+  const resultObject = {
+    singleChain: [],
+    multiChain: [],
+  };
+  try {
+    const url = 'https://unstoppabledomains.com/api/uns-resolver-keys';
+    const response = await fetch(url);
+    const data = await response.text();
+    const currencyArray = parseKeysArray(JSON.parse(data).keys);
+    currencyArray.forEach(function (crypto) {
+      if (crypto.includes('.version.')) {
+        resultObject.multiChain.push(crypto.split('.')[1]);
+      } else {
+        resultObject.singleChain.push(crypto.split('.')[1]);
+      }
+    });
+    return resultObject;
+  } catch {
+    return resultObject;
+  }
+}
```
These two methods are adding centralized services to our code, and exposing our user's ip addresses to unstoppable domains. Is there any alternative to http requests to retrieve this info? @Gudahtt can I get your take on this as well?

bbehrman10 replied:
As of now, our only alternative to this would be to hardcode constants for the TLDs and Supported Currencies - so each time those change, an update to the codebase would be required

bbehrman10 replied:
if it helps, all of the data contained within the API is open source which is less centralized

brad-decker reviewed (changes requested):
Before any further review will happen:

1. Please go through and update variable names like 'object' to something of meaning for reviewers and future maintainers of this code. Naming things is hard, but it is absolutely vital for legibility and understanding code and it is impossible to review when combined with the second bullet point:
2. Add jsdocs to every method you are introducing that describe what it does, and also any inline docs that help to explain your intent with the changes being made. 
3. Analyze my request to remove web requests for viable TLDs for unstoppable domains, as well as keys. 

Thank you for the contribution and I hope we can work through these changes and find a way to get this landed.

bbehrman10 committed:
add MetaMask requested changes, add JSDOCS comments

bbehrman10 committed:
fixes some unit testing issues

bbehrman10 committed:
attempts to resolve an undefined payload issue

bbehrman10 committed:
checks paylaod for undefined and null

bbehrman10 committed:
another attempt at resolving the testing issue

Gtonizuka commented:
Possible duplicate of [#7258](https://github.com/MetaMask/metamask-extension/pull/7258)

jim-unstoppable commented:
> Possible duplicate of [#7258](https://github.com/MetaMask/metamask-extension/pull/7258)

This is an updated implementation reflecting support for our most recent registry inclusive of Polygon based NFT Domains. This version uses our most recent and established libraries, so I think the old one is no longer applicable

bbehrman10 commented:
@brad-decker - We can submit a change and move the data from the endpoint to a constants file. If that's okay with you we can do that, and in the coming weeks / months if there needs to be an update to those constants (such as if UD releases more token support) just submit a PR with only changes to that constants file. In this way the integration can stay up to date without the need of that REST endpoint. How does that sound? Alternatively we're also comfortable if you are to keep that open source endpoint to leave it all server side.

brad-decker commented:
Open source does not mean decentralized, and does not mitigate the risk of exposing our user's ip addresses to UD. Our preference would be for your team to create a npm module that contains the constants necessary, linked to an open source repository that can be validated by our team to contain neutral code/constants/etc. Then your team may update those constants and publish new versions which can then be upgraded in our codebase and a reviewer can validate the changes.

Just so you're aware, once the requested changes have been made i'll be doing further review of your integration into the send slice. I would leave additional inline comments everywhere where they may be some context needed for proper understanding of what the code is doing.

We audited the ENS integration and no centralized services are hit to perform the reverse lookup or resolutions. We would expect UD to meet the same standards to be included in the code.

bbehrman10 committed:
swaps out API with new NPM package

bbehrman10 committed:
adds UNS tests and cleans up some send.js logic

bbehrman10 committed:
adds a custom timeout to allow for UD calls

bbehrman10 commented:
@brad-decker quick question. Our unit tests we wrote are passing locally but when running as part of the suite on CircleCI we are getting an 'ENETUNREACH' - have you experienced this before? How could we alter our tests to prevent this from occuring

brad-decker commented:
@bbehrman10 unit tests should not hit the network, you can't test network integrations with unit tests. You'll need to mock out those network requests. The easiest way to do that, if the network request happens in the background, is to mock the background action call such that the code that makes the network request is never reached. If the network requests happen in the UI you'll need to use nock or similar to replace those network calls. You can do nock in the background as well if you'd prefer the background call to occur naturally in tests...

ultimately though if you replace those network calls with flat files from a npm module as discussed above you should be good. Other than retrieving TLDS and keys what other network calls are made that aren't to the ethereum client?

bbehrman10 commented:
@brad-decker gotcha that makes sense. Okay we will remove those network calls altogether. Thanks!

bbehrman10 committed:
changes resolution provider and removes standalone UNS test file

bbehrman10 committed:
updates a comment in the utility file

bbehrman10 committed:
recommiting for testing

bbehrman10 commented:
@brad-decker hey! We are ready for you to take another look. We have moved the TLDs and Resolver Keys to this NPM package: https://www.npmjs.com/package/@unstoppabledomains/tldsresolverkeys - Docs here: https://github.com/unstoppabledomains/tlds-and-resolver-keys-package. We have removed all network calls aside from the same calls to infura that metamask is making.

brad-decker commented:
@bbehrman10 I think it is best for us to split up some of this work. I think a good first PR would be renaming all of the ENS specific selectors, state keys, components to use the generic term domain like you have done in this PR. We can get that landed and then layer on some of this work. I'm going to still do a full review of this pr but I know I'm going to ask for that, so if you wanted to get a jump start I think it would be worthwhile.

bbehrman10 commented:
@brad-decker sounds good! If you can get us those other comments ASAP that would be great.

bbehrman10 committed:
combines UNS and ENS basic resolution into one module and generalizes their state to domains rather than their individual names

bbehrman10 committed:
meaningless change so circleci will run the test

bbehrman10 commented:
@brad-decker - So we combined the resolution modules for ENS and UNS, and put them into a single state slice called domainState. Something interesting is going on with the testing. So when we run the unit tests they all do pass, but the actual testing process timeouts before it prints out the summary. We are also passing both chrome and firefox e2e tests. You'll see these behaviors on the circleCI tests as well. Should we disregard this timeout issue? We've pushed these changes for when you're ready to take a look. Thanks so much!

Mrtenz committed:
Update Jest config to avoid transpiling dependencies (#16068)

* Use stable version of Jest

* Update Jest config

ryanml committed:
Fixing Identicon import paths (#16101)

brad-decker commented:
@bbehrman10 what was the status of splitting off some of this work to make the review less complex?

ryanml committed:
Fixing forgot password action after wallet creation (#16156)

bbehrman10 commented:
@brad-decker - we made a new PR for the name changes requested: https://github.com/MetaMask/metamask-extension/pull/16166

jpuri committed:
Updating controller dependency (#16033)

* Updating controller dependency

* fix

* fix

* fix

* fix

* fixes

* Lavamoat auto

* Update URLs for phishing detection testcase

* update lavamoat files

* call phishingController.test synchronously again

* bump @metamask/controllers to v32.0.1

* lint

* update policy files

* bump controllers version again

* modify update phishing list strategy

* revert back to use isOutOfDate, but without blocking substream

* possible way to fix e2e tests?

* enable testing

* Remove promise return from setupController in background.js, as it is no longer used

* Ensure updatePhishingLists is called in MM contrller constructer, so that phishing lists are updated right away

Co-authored-by: seaona <mariona@gmx.es>
Co-authored-by: Alex <adonesky@gmail.com>
Co-authored-by: Dan Miller <danjm.com@gmail.com>

ryanml committed:
Remove ETH badge from NetworkAccountBalanceHeader when on non-ETH net… (#16102)

* Remove ETH badge from NetworkAccountBalanceHeader when on non-ETH networks

* Network specific icons, or fallback, in the network-account-balance-header

* Update snapshots

* Code cleanup

Co-authored-by: Dan Miller <danjm.com@gmail.com>

seaona committed:
[e2e] Fix Dapp interactions flaky test on chrome (#16282)

ryanml committed:
Enable "Add Popular Network" feature by default for all users (#16172)

* Enable "Add Popular Network" feature by default

* Fixing e2e tests

* Attempt to fix e2e tests

* Revert "Attempt to fix e2e tests"

This reverts commit d88e8944b8df8f7c3075753f8d8b3480439f8e30.

* Get e2e tests passing

Co-authored-by: Dan J Miller <danjm.com@gmail.com>

seaona committed:
Changelog last updates with cherry-picked PRs (#16287)

* Last changelog updates with recently added PRs

* Remove snaps-skunkworks from Changelog

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

* Fix snaps issue that request eth_accounts permission

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

* Move flask PR to fixed section

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

seaona committed:
Remove empty files

seaona committed:
Remove duplicate RC entry 10.20.0

seaona committed:
lavamoat

FrederikBolding committed:
Improve transaction confirmation page performance (#16205)

* Reduce the amount of calls to insight snaps

* Add dep

* Update LavaMoat policies

* Fix selectors and revert changes to hook

* Remove dep

* Revert "Update LavaMoat policies"

This reverts commit 7469c94e0e7c52eb877766a96005cc559c9d8d20.

PeterYinusa committed:
E2e test fixtures (#16061)

* remove state.json files

* move file

* Update Readme

* Create fixture builder

* load test fixture

* remove redundant method

* update snap tests

* update stats tests

* update extension tests

* update extension tests

* Update fixture data

* snap test dapp connection

* Update fixture data

* add onboarding fixture

* use onboarding fixture

* reuse import account vault

* remove unnecessary use of class

* use fixture builder in new tests

* switch to function

* update default fixture

* update default fixture

* update test

* update 1559 test fixttures

* update 1559 test fixtures

* update 1559 test fixtures

* dismiss 3box whats new

* remove redundant code

* move docs

* remove unused code

* token detection

* use default timeout

* remove redundant code

* Update fixture builder

hide `Protect your funds` dialog
remove browser environment
remove default network details
hide dismiss seed backup reminder
recursively merges fixture data

* add token to tokencontroller

* remove network details

* add missing identities to preference controller

* remove duplicate properties

* update bip-32 to use fixturebuilder

* alphabetise snap permissions

* update get snaps to use fixturebuilder

* Update test-snap-bip-32.spec.js

wait for window

* add popular network state

* update test

* lint

seaona committed:
lavamoat

dan437 committed:
Add e2e tests for Swaps notifications (#16202)

dan437 committed:
Exclude E2E tests for swaps until they are stabilised (#16303)

Gudahtt committed:
Merge pull request #16036 from MetaMask/Version-v10.21.0

Version v10.21.0

Gudahtt committed:
Merge remote-tracking branch 'origin/develop' into sync-master

* origin/develop: (79 commits)
  Exclude E2E tests for swaps until they are stabilised (#16303)
  Add e2e tests for Swaps notifications (#16202)
  E2e test fixtures (#16061)
  Improve transaction confirmation page performance (#16205)
  [MV3] Fix e2e `encrypt-decrypt` test for MV3 test build (#16264)
  Fix e2e incremental-secuirty test for mv3 build (#16265)
  Enable "Add Popular Network" feature by default for all users (#16172)
  [e2e] Fix Dapp interactions flaky test on chrome (#16282)
  [FLASK] Allow snaps insights to show on regular EOA transactions (#16093)
  TextFieldBase updates after design reveiw (#16201)
  Adding `TextField` component (#16105)
  Remove ETH badge from NetworkAccountBalanceHeader when on non-ETH net… (#16102)
  Updating controller dependency (#16033)
  Fix inputs that should allow decimals (#16254)
  Chromedriver v106 (#16213)
  [FLASK] Clear snap state on restore (#16228)
  [FLASK] Fix flaky BIP-32 E2E test (#16255)
  Continue converting tests from enzyme to @testing-library/react (#16175)
  Fix/16207/text component storybook (#16214)
  ci: fix stats-module-load-init step (#16196)
  ...

jpuri committed:
Making permission and approval controller methods idempotent (#15848)

PeterYinusa committed:
[E2E] Restore tests (#16298)

* add restore test

* rename spec

rekmarks committed:
Remove 3box feature and delete ThreeBoxController (#14571)

* Remove 3box feature and delete ThreeBoxController

Lint locale messages

lavamoat policy updates

* Restore 3Box user trait with value `false`

The 3Box user trait has been restored and hard-coded as `false`. This
ensures that users don't get stuck in our metrics as having this trait.

A deprecation comment has been left in various places for this trait.

* Remove unused state

* Remove additional 3box-related things

* Run `yarn-deduplicate`

* Restore migration that was lost while rebasing

* Remove obsolete override

* Remove additional unused resolutions/dependencies

* Update LavaMoat policies

* Remove obsolete security advisory ignore entries

* Remove 3Box fixture builder method

* Update unit tests

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

Gudahtt committed:
Merge branch 'develop' into sync-master

Gudahtt committed:
Merge pull request #16309 from MetaMask/sync-master

Sync v10.21.0 with develop

Gudahtt committed:
Increase provider rate limited to 5 minutes (#16308)

The metric event for provider methods has been increased from 1 minute
to 5 minutes. This will reduce the event rate, ensuring a maximum of
one event is tracked for each method call every five minutes.

Gudahtt committed:
Increase provider rate limited to 5 minutes (#16308)

The metric event for provider methods has been increased from 1 minute
to 5 minutes. This will reduce the event rate, ensuring a maximum of
one event is tracked for each method call every five minutes.

metamaskbot committed:
Version v10.21.1

Gudahtt committed:
Update changelog

adonesky1 committed:
remove unused zh locales directory in favor of zh_CN (#16322)

Gudahtt committed:
Merge pull request #16328 from MetaMask/Version-v10.21.1

Version v10.21.1 RC

adonesky1 committed:
ensure that we default users with browser language code 'zh' to the supported 'zh_CN' language code (#16329)

legobeat committed:
ci: temporarily downgrade ubuntu to 20.04 (#16320)

ryanml committed:
Fix "Learn more" link on Connected Account notification (#16339)

kumavis committed:
[LavaMoat] improve policy generation performance in dev workflow (#16338)

* npm scripts - rename lavamoat:background to lavamoat:webapp

* lavamoat - generate webapp policy - add devMode flag to run in node for speed

* lavamoat - generate webapp policy - dont lint fences

* lint fix

seaona committed:
Updating chromedriver to last stable version (#16310)

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

seaona committed:
Merge branch 'master' into develop

Gudahtt committed:
Merge pull request #16351 from MetaMask/sync-master

Sync develop with master v10.21.1

seaona committed:
Run update-changelog script

mirjanaKukic committed:
added new tests for editing and deleting contact from address book (#16014)

lint fix

fix

added fixture builder

FrederikBolding committed:
[FLASK] Fix crash on snaps settings page if snap has no permissions (#16354)

ryanml committed:
Fix Secret Recovery Phrase video clipping issue for smaller viewports (#16345)

adonesky1 committed:
Fix issue where ens resolution stops working after sw is terminated (#16335)

* fix issue where ens resolution stops working after sw is terminated

seaona committed:
Merge branch 'develop' into Version-v10.22.0

tmashuang committed:
Continue converting tests from enzyme to @testing-library/react (#16249)

dan437 committed:
Enable and stabilise e2e tests for Swaps, add debouncing (#16326)

seaona committed:
Merge branch 'develop' into Version-v10.22.0

georgewrmarshall committed:
Adding HelpText component (#16293)

* Adding initial HelpText component

georgewrmarshall committed:
Adding Label component (#16203)

* Adding Label component

* doc fixes

* Adding label wrapping input

segun committed:
Show error message if service worker did not load (respond to the UI) (#15774)

* Show error message if service worker did not load (respond to the UI)
after 1 minute.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

* Remove console.log

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

* New Error message design

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

* Fix tests

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

* Use lastTimeStamp instead of keeping track of message ids

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

* Do not initial channe every second.

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

* New logic to check if SW is stuck

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

Signed-off-by: Akintayo A. Olusegun <akintayo.segun@gmail.com>

garrettbear committed:
button base housekeeping (#16336)

* button base housekeeping

* add pixel size to description

* snapshot update

NidhiKJha committed:
tag component housekeeping (#16355)

* tag component housekeeping

* added snapshot testing

legobeat committed:
build: run scss pipelines synchronously (#16301)

This resolves a race condition in live-reload when editing scss files.

FrederikBolding committed:
[FLASK] Fix some issues with installing snaps that request eth_accounts(#16365)

NEllusion committed:
Fix the rendering of certain domain names in connected-sites list (#16074)

* Fix rendering origins in connected-sites list

NEllusion committed:
Fix the rendering of certain domain names in connected-sites list (#16074)

* Fix rendering origins in connected-sites list

PeterYinusa committed:
Update PULL_REQUEST_TEMPLATE.md (#16367)

dependabot[bot] committed:
Bump @metamask/providers from 10.0.0 to 10.2.0 (#16361)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

NiranjanaBinoy committed:
Fix: To differentiate between which chrome alarm is triggered (#16360)

adonesky1 committed:
Fix issue with awaiting swaps page (#16344)

* fix issue with awaiting swaps and smart-transaction-status pages when service worker dies

dragana8 committed:
[Bug]: Balance does not update on duplicate chainId network #13690 (#14245)

jpuri committed:
Fix approve transaction call in _onBootCleanUp (#16382)

jpuri committed:
Fix approve transaction call in _onBootCleanUp (#16382)

tmashuang committed:
Continue converting tests from enzyme to @testing-library/react (#16373)

* Add transaction activity log component

* Remove duplicate tx activity log snapshot.

* Convert Identicon test to tlr.

* Convert Metafoxlogo test to tlr.

* Convert Reveal Seed Phrase test to tlr.

* Consolidate and convert Send Footer test to tlr.

* Convert Settings test to tlr.

* Consolidate and convert security tab test to tlr.

* Convert null selectedOption to empty string, and add test id to Dropdown component.

* Add Send state to mock-state

* Lint mock-state.json

legobeat committed:
Revert "ci: temporarily downgrade ubuntu to 20.04" (#16374)

This reverts commit bc6c6c4de4af7da210269700c5f6c417d27d3030.

seaona committed:
Changelog updated

seaona committed:
Fix lint

danjm committed:
Merge pull request #16395 from MetaMask/Version-v10.21.2

Version v10.21.2

Commit:
Token allowance screen updated (#16157)

BelfordZ committed:
Change readme for running e2e (#16386)

georgewrmarshall committed:
HelpText propType fix (#16402)

jpuri committed:
Persisting segment events in MetaMetricsController store (#16198)

amerkadicE committed:
Update address component on transaction data screen (#15888)

* Update address component on transaction data screen

E2E test fix

Fix tests

* Metadata accounts

* Remove test resulsts

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

FrederikBolding committed:
[FLASK] `snaps-monorepo@0.23.0` (#16394)

* snaps-monorepo@0.23.0

* Regen LavaMoat policies

* Update iframe-execution-environment

* Fix tests

* Bump luxon

* Fix lint

* Update test-snaps

* Bump post-message-stream

* Fix E2E

seaona committed:
Sync master and update changelog

seaona committed:
Sync develop with master v10.21.2 (#16399)

* Fix approve transaction call in _onBootCleanUp (#16382)

* Changelog updated

* Fix lint

Co-authored-by: Jyoti Puri <jyotipuri@gmail.com>
Co-authored-by: Dan J Miller <danjm.com@gmail.com>

Commit:
Add improved token allowance experience toggle (#16291)

FrederikBolding committed:
Parallelize E2E tests to cut CI time in half (#16417)

* Experiment with parallellizing E2E

* Fix lint

* Try parallelism 8

* Apply suggestions from code review

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

NiranjanaBinoy committed:
Fix to direct the token detection settings search to the correct row (#16407)

jpuri committed:
Make event tracking idempotent using unique messageId (#16267)

dan437 committed:
Enable new networks for Swaps (#16396)

segun committed:
Fix orphaned data warning (#16413)

david0xd committed:
[FLASK] Add snap cronjobs (#16239)

* Add Cronjob controller configuration

* Add Cronjob permission icon and description

* Add lint fix changes

* Add missing action to the allow-list

* Fix permission

* Fix icon and message

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

GuillaumeRx committed:
[FLASK] Catch and display errors in snaps insight (#16416)

* snaps insight error catching and error state

* revert confirm-transaction-base changes

* display empty state if data is null

* add loading state

* reset errors and loading state on call

* update hasNoData

* update length check in component

mikesposito committed:
chore: add bundle size diff to metamaskbot (#16098)

adnansahovic committed:
Warning on a Send transaction request (#16220)

adnansahovic committed:
Warning on a Send transaction request (#16220)

Gudahtt committed:
Remove obsolete builds (#16430)

The `brave` and `opera` builds are obsolete; in practice we have been
using the Chrome build for those browsers, because they are based upon
Chromium and are compatible with the Chrome extension API.

PeterYinusa committed:
E2e remove redundant code (#16432)

* remove unused code

* remove unused code

darkwing committed:
BETA - Update logo imagery (#16304)

* BETA - Update logo imagery

* Restore basic metamask rotating fox

* Adding story and updating colors

* Add beta to following logo

* New, square images

* Add 512 icon

Co-authored-by: georgewrmarshall <george.marshall@consensys.net>

tmashuang committed:
Continue converting tests from enzyme to @testing-library/react (#16401)

PeterYinusa committed:
refactor sentryHooks object (#16435)

garrettbear committed:
Feat/16290/add button component (#16305)

* add button component

* add all button props

* update tests

* add button type prop

* fix base button size const

* add href prop to button base

* Update ui/components/component-library/button/README.mdx

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button/README.mdx

Co-authored-by: George Marshall <george.marshall@consensys.net>

* update tests

* Update ui/components/component-library/button-primary/button-primary.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button/README.mdx

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button/button.stories.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button/button.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button/button.stories.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* update button props on readme

* linting issue fix

Co-authored-by: George Marshall <george.marshall@consensys.net>

garrettbear committed:
Feat/15088/add button icon (#16277)

* 15088: add button icon

* button icon story updates

* add primary type

* update button icon docs

* add test

* button icon updates

* button icon and test updates

* button icon border radius and test update

* remove padding prop

* button icon updates

* Update ui/components/component-library/button-icon/button-icon.stories.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button-icon/button-icon.stories.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/component-library/button-icon/button-icon.stories.js

Co-authored-by: George Marshall <george.marshall@consensys.net>

* add aria label for storybook demo

Co-authored-by: George Marshall <george.marshall@consensys.net>

darkwing committed:
BETA - Update verbiage for success messages after onboarding (#16343)

darkwing committed:
BETA - Update verbiage for success messages after onboarding (#16343)

NidhiKJha committed:
Added Picker Network Component (#16340)

* added file structure for picker-network

* updated accessory prop with props

* added tests and documentation to picker network

* updated picker network classnames

* updated custom tests for picker network

* updated css in picker network

* updated readme and stories

* added snapshot testing

* changed behaviour to button

* updated snapshot for button

* updated label as per src

georgewrmarshall committed:
Update `TextFieldBase` to include InputComponent prop (#16422)

* Update TextFieldBase to include InputComponent

* Updating docs

* Adding test

* Update to docs

* Lint fixes

* Small doc fixe to remove some unneded spaces

NidhiKJha committed:
Added Tag Url Component  (#16159)

* added tag url file structure

* added tag url component

* updated args in tag url

* updated avatar favicon and button link in Tag url

* updated Text component in Tag URL

* added tests to the tag url component

* fixed typos

* replaced buttonbase with button link

* added lock icon to the tag-url component

* updated typo

* renamed style and prop names

* updated prop name and  tests

* fixed name for tag in readme

* fixed test statements

* updated prop names

* updated story

* fixed typo in test file

* updated lock icon rendering and icon color

* updated lock icon prop

* updated README

* updated tests for tag url

* added snapshot testing

* updated Readme for tag url

* updated label as per src

Co-authored-by: Garrett Bear <gwhisten@gmail.com>

seaona committed:
Changelog for 10.22.0 (#16436)

* Changelog for 10.22.0

* Update CHANGELOG.md Fix crash after restoring MetaMask if you had snaps installed

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

* Update CHANGELOG.md Fix a problem with displaying snaps transaction insight data

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

* Update CHANGELOG.md Improve transaction confirmation page performance

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

VSaric committed:
Replaced addresses by the address component on SignTypedData v4 signatures (#16018)

* Replaced addresses by the address component on SignTypedData v4 signatures

* Fixing signature-request e2e tests

* Modified scss file for signature-request message

* Using address component for rendering the addresses and bold label where hex address is not valid

* Modify the address component

* Added proper BEM syntax for class names and used Box and Typography

* FIxing e2e tests

* Commited requested changes from George and added storybook

* Review requested changes

* Created new component for rendering data in signature-request-message.js

* Fixing proper usage for getAccountName and getMetadataContractName selectors

* Fixing e2e tests

seaona committed:
Merge branch 'master' into Version-v10.22.0

PeterYinusa committed:
Upgrade geckodriver to latest version, run ci tests with Firefox v106.0.4 (#16383)

* update geckodriver

* update ff binary in ci

* dedupe

danjm committed:
Call onbootcleanup at the end of the tx controller constructor (#16449)

* Call onbootcleanup at the end of the tx controller constructor

* Update app/scripts/controllers/transactions/index.js

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

danjm committed:
Call onbootcleanup at the end of the tx controller constructor (#16449)

* Call onbootcleanup at the end of the tx controller constructor

* Update app/scripts/controllers/transactions/index.js

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

gzliudan committed:
docs: fix obsolete link of nvm in README.md (#16238)

Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

georgewrmarshall committed:
`TextField` component updates (#16424)

* Updating showClearButton prop name and making component dumber

* Docs update

* Adding missing proptype

* Fixing casing on tests

* Replacing clear button placeholder with ButtonIcon and docs update

* Fixing linting issues

* Adding note about controlled only for showClearButton to work and fixing some tests

* Updating test to include controlled testing setup function for clearButton tests

adonesky1 committed:
fix chainId validation on network form (#16452)

adonesky1 committed:
fix chainId validation on network form (#16452)

seaona committed:
Add Changelog entry for fix Network form by validating chainId

Gudahtt committed:
Merge pull request #16353 from MetaMask/Version-v10.22.0

Version v10.22.0

seaona committed:
Merge branch 'master' into master-sync

FrederikBolding committed:
[FLASK] Fix an issue with updating snaps that have caveat permissions (#16473)

PeterYinusa committed:
[e2e]: log exceptions when tests fail (#16453)

* log exceptions inc background

* show full stacktrace

* use driver api

* lint

Gudahtt committed:
Merge pull request #16476 from MetaMask/master-sync

Sync develop with master v10.22.0

digiwand committed:
MV3: Update service worker restart logic and keep-alive logic for dapp support (#16075)

* dapp: add debug statements

* dapp: add retry logic [debug]

* dapp: keep SW alive on rpc request

* Revert "dapp: add debug statements"

This reverts commit ea21786f7f66c712eea02405cd68fe925d227ffa.

* dapp: try to set up ext streams asap on reset

* dapp: apply keep alive logic to phishingPageStream

* dapp:put keep-alive logic behind isManifestV3 flag

* Re-activate streams after a period of service worker in-activity

* dapp: rm extra function

* dapp: update phishing onDisconnect

* dapp: fix eslint missing global chrome

* add EXTENSION_MESSAGES const

* use EXTENSION_MESSAGES more generic comment

* update comment

* dapp: clean timeout and interval

* Fix DAPP action replay

* execute DAPP action replay for only MV3

* fix

* fix

* fix

* comment out DAPP action replay code

* fix

* fix

* fix

* scripts/background: use browser polyfill

* Revert "scripts/background: use browser polyfill"

This reverts commit 2ab6234d11b3b11e10dd993d454eeaad63bfc886.

* scripts/background: use browser polyfill

* script/background: check lastError

* dapp: use EXTENSION_MESSAGES

* scripts/background: send ready msg to all tabs

* dapp: update onMessage handler comment and name

* dapp: return values onMessage listener
see: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage#addlistener_syntax

* dapp: mv onMessage listener

* dapp: add onMessage setupPhishingExtStreams

* dapp: rn reset -> destroy streams

* dapp: rn reset -> destroy for phishing streams

* dapp: clean comment

* dapp: rm unused comments
planning to be readded in follow-up PR: #16250

* dapp: onMessage return Promise|undefined

* dapp:clean: add missing undefined return type

* dapp: use new checkForErrorAndLog for Chrome API
handy stackoverflow: https://stackoverflow.com/a/28432087/4053142

* dapp:fix: return tabs.query result

* dapp:eslint: return undefined
fix Expected to return a value at the end of arrow
function.eslintconsistent-return

* background: do not query tabs w/out url

* background: rm Could not establish... catch
- no longer needed after improved tabs query

* dapp:clean: rm unused checkForError... for now...

* dapp: prevent setupExtensionStreams called twice
- calling connect will trigger disconnect and may cause issues
- only setup streams if they are not connected

* dapp: handle onDisconnect lastError
- throwing errors from contentscript will break the dapp, so only warn
- not handling lastError when it's found will also break the dapp

* background: update tabs.query url comment

* background: update tabs.query url comment 2

* dapp: fix SW restart for multi dapp support
- ref: https://stackoverflow.com/a/54686484/4053142

* dapp:clean: rm extra "." from console.warn

* clean: comments for dapp and background

* Adding catch block (#16454)

* fix: FireFox provider injection

* lib/util: fix invalid checkForErrorAndWarn export

* bg: add explanation for tabs.sendMessage catch

* dapp: add browser-runtime.utils

* runtime.utils: add checkForLastErrorAndLog

Co-authored-by: Jyoti Puri <jyotipuri@gmail.com>
Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

adnansahovic committed:
Added a 'Verify contract details' link to SetApprovalForAll confirmation screens (#15756)

andy0130tw committed:
[i18n:zh_tw] Sync and update translation for zh_TW (#11212)

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

NidhiKJha committed:
added 50% border radius support to the Box component (#16486)

* added 50% border radius support to the Box component

* added height and width

* fixed lint issue

garrettbear committed:
Feat/16185/button primary housekeeping (#16457)

* ButtonPrimary housekeeping

* use sizes

* update test and danger type

brad-decker committed:
upgrade gulp-autoprefixer (#16439)

Gudahtt committed:
Revert "Merge pull request #16353 from MetaMask/Version-v10.22.0"

This reverts commit 308033edbf0fcda8d15c0c0c3b230463983232c3, reversing
changes made to b8d71d6d1d23d4a914201fd2b1427033616703b4.

metamaskbot committed:
Version v10.22.1

Gudahtt committed:
Restore v10.22.0 changelog

danjm committed:
Prevent user from editing a contract interaction created by a dapp (#16498)

* Prevent user from editing a contract interaction created by a dapp

* Code cleanup

* Fix e2e test selector

* Fix e2e test

* Fix e2e test

* Update snapshot

danjm committed:
Prevent user from editing a contract interaction created by a dapp (#16498)

* Prevent user from editing a contract interaction created by a dapp

* Code cleanup

* Fix e2e test selector

* Fix e2e test

* Fix e2e test

* Update snapshot

Gudahtt committed:
Update changelog

Gudahtt committed:
Ignore advisary related to 3Box

3Box was removed in v10.22.0, so this advisory will be obsolete in the
subsequent release, which is being prepared already.

Gudahtt committed:
Fix e2e test

The new e2e test added for the bug fix being released here has been
updated to use the old e2e test fixtures, rather than the fixture-
builder. The fixture builder is not available on this branch yet.

georgewrmarshall committed:
Adding `TextFieldSearch` component (#16296)

* Adding TextFieldSearch component

* Updating docs and stories

* Moving controlled test into testing utils

* Fixing spelling in prop types af => of

Gudahtt committed:
Merge pull request #16504 from MetaMask/Version-v10.22.1

Version v10.22.1 RC

Gudahtt committed:
Revert "Fix e2e test"

This reverts commit 69ccb37cb957447709ee80508a897c098e691f6f.

Gudahtt committed:
Revert "Ignore advisary related to 3Box"

This reverts commit 3da175cf0c94150ead2b2d2c2349a67832d6d25d.

Gudahtt committed:
Revert "Revert "Merge pull request #16353 from MetaMask/Version-v10.22.0""

This reverts commit d91ce9c4b5044e906f254516a95d87ed74f794d6.

georgewrmarshall committed:
Adding some propType fixes to `TextFieldBase` (#16508)

* Adding some propType fixes

* lint fixes

digiwand committed:
Jest: Add browser.runtime (webextension-polyfill) utils tests (#16483)

* jest: add browser-runtime.utils tests

* browser-runtime.utils: rm checkForLastErrorAndWarn
- will be removing use in https://github.com/MetaMask/metamask-extension/pull/16488

Gudahtt committed:
Update `@metamask/controllers` to v33 (#16493)

The controllers package has been updated to v33. The only breaking
change in this release was to rename the term "collectible" to "NFT"
wherever it appeared in the API.

Changes in this PR have been kept minimal; additional renaming can be
done in separate PRs. This PR only updates the controller names,
controller state, controller methods, and any direct references to
these things. NFTs are still called "collectibles" in most places.

metamaskbot committed:
Version v10.22.2

adnansahovic committed:
Added "What's New" Notification for Improved token allowance experience (#16465)

Gudahtt committed:
Update changelog

garrettbear committed:
button secondary housekeeping (#16495)

* button secondary housekeeping

* add snapshot

* add export constants

Gudahtt committed:
Update SmartTransactionsController to v3 (#16511)

This update drops support for Node.js v12, and it includes a bug fix
for a deadlock encountered when using swaps on unsupported testnets.

Fixes #16479

Gudahtt committed:
Update SmartTransactionsController to v3 (#16511)

This update drops support for Node.js v12, and it includes a bug fix
for a deadlock encountered when using swaps on unsupported testnets.

Fixes #16479

danjm committed:
Configure smart transactions controller to support goerli (#16500)

danjm committed:
Configure smart transactions controller to support goerli (#16500)

brad-decker committed:
persist user traits for comparison (#16506)

darkwing committed:
Remove unnecessary await (#16482)

seaona committed:
policies + yarn

adnansahovic committed:
Modified chain_id value(MetaMetrics), in the Custom Network Added event (#16426)

dan437 committed:
Add mocks for E2E tests (#16527)

* Mock token import

* Add mocks, improve stability

* Update LavaMoat policies

The policies became outdated recently due to undetected conflicts
between PRs.

* Use veryLargeDelayMs, refactoring

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

dan437 committed:
Add mocks for E2E tests (#16527)

* Mock token import

* Add mocks, improve stability

* Update LavaMoat policies

The policies became outdated recently due to undetected conflicts
between PRs.

* Use veryLargeDelayMs, refactoring

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

seaona committed:
yarn dedupe + policies + changelog

seaona committed:
Update CHANGELOG.md entry

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

darkwing committed:
Beta: Update long logo to new imagery (#16505)

darkwing committed:
BETA - Add beta banner to all screens (#16307)

* Add beta home banner to home screen

* Move the beta home notification to the app-header

* Updates to formatting

* Add beta home banner to home screen

* Move the beta home notification to the app-header

* Updates to formatting

* Update ui/components/app/app-header/index.scss

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/app/app-header/index.scss

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Update ui/components/app/app-header/index.scss

Co-authored-by: George Marshall <george.marshall@consensys.net>

* Move banner to top of page

* Move to own folder, update styles

* Swith to BOX component

* Address feedback

* Add tests

* Update name of component

* Fix tests

* Fix proptype errors

* Fixups

* Remove unrelated changes

* Remove unwanted string changes

* Update beta component name and text

* Update mock data

Co-authored-by: George Marshall <george.marshall@consensys.net>

dependabot[bot] committed:
Bump @metamask/design-tokens from 1.9.0 to 1.11.0 (#16515)

Bumps [@metamask/design-tokens](https://github.com/MetaMask/design-tokens) from 1.9.0 to 1.11.0.
- [Release notes](https://github.com/MetaMask/design-tokens/releases)
- [Changelog](https://github.com/MetaMask/design-tokens/blob/main/CHANGELOG.md)
- [Commits](https://github.com/MetaMask/design-tokens/compare/v1.9.0...v1.11.0)

---
updated-dependencies:
- dependency-name: "@metamask/design-tokens"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

darkwing committed:
BETA - Updating welcome message (#16489)

seaona committed:
[e2e] Fix RPC flaky test (#16530)

seaona committed:
[e2e] Fix RPC flaky test (#16530)

bowensanders committed:
[FLASK] Update e2e tests for new react test-snaps page (#16324)

* Creates updated e2e tests compatible with the new react based test-snaps page.

seaona committed:
policies update

paulbuechner committed:
fix(i18n): de lightTheme translation (#16517)

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

georgewrmarshall committed:
Component library adding global index and other housekeeping (#16441)

* Adding global index.js file

* Removing style prop from button base

* Fixing comment on generate-icon-names.js file

* Re-ordering sass imports to atomic structure

* Updating component paths code examples of MDX docs

seaona committed:
Update package json

seaona committed:
yarn deduplicate

seaona committed:
policies (hopefully last one)

danjm committed:
Merge pull request #16516 from MetaMask/Version-v10.22.2

Version v10.22.2 RC

danjm committed:
Merge remote-tracking branch 'origin/develop' into master-sync

Gudahtt committed:
Merge pull request #16548 from MetaMask/master-sync

Master sync following v10.22.2

adnansahovic committed:
Skip the first token allowance screen if dapp proposing spending cap is 0 (#16502)

Commit:
Add transaction security check toggle (#16271)

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

Gudahtt committed:
Fix crash on e2e test failure (#16556)

If an e2e test fails on a website, the driver will crash on this line
because it can't find a reference to `stateHooks`. That variable is
only set in our UI, not on other sites.

digiwand committed:
MV3: support Service Worker restart for phishing warning pages (#16488)

* dapp: fix phishing Could not establish connection

* dapp: only call phishing keep alive timer on load

Co-authored-by: Jyoti Puri <jyotipuri@gmail.com>

Co-authored-by: Jyoti Puri <jyotipuri@gmail.com>

danjm committed:
Ensure prod beta build is created when merging to master (#16557)

* Ensure prod beta build is created when merging to master

* Fix: remove negation

* Update .circleci/config.yml

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

georgewrmarshall committed:
Updating all background-default hovers colors (#16519)

georgewrmarshall committed:
Updating network colors with design tokens (#16543)

* Updating network colors with design tokens

* Style update to flask page

NidhiKJha committed:
Updated Text style in Tag Url (#16570)

* fixed truncate in tag url

* updated snapshot testing

Co-authored-by: Garrett Bear <gwhisten@gmail.com>

digiwand committed:
Minor scripts/ui.js file cleanup (#16566)

* clean:ui: mv comments

* clean:ui: fix typo recieved -> received

* clean:ui: rn handle -> keepAliveInterval

* clean:ui: rn timeoutHandle -> ackTimeoutToDisplayError

* clean: replace "Xs" w/ variable name

* clean:ui: rm duplicate comment

* add EXTENSION_MESSAGES.CONNECTION_READY const

* clean:ui: bit more clean up

georgewrmarshall committed:
Removing storybook tests (#16577)

amerkadicE committed:
Fix message on sign typed data v4 screen (#16552)

Co-authored-by: Frederik Bolding <frederik.bolding@gmail.com>

georgewrmarshall committed:
Adding static icon names to test env file (#16078)

* Storing the icon name env var as a string and parsing to use with components

* Moving icon env vars to jest specific env.js file

* Updating snapshots

NidhiKJha committed:
updated background color for picker network (#16466)

* updated background color for picker network

* updated color for pressed state

* fixed linting error

* updated snapshots

* removed borderstyle

* update color

* updated snapshot

* added truncate to the text

* updated snapshot

NiranjanaBinoy committed:
Updating tokensChainsCache[chainId].data to object from array in user state (#16535)

garrettbear committed:
Feat/16507/button link housekeeping (#16518)

NidhiKJha committed:
updated responsive props for border radius (#16510)

* updated responsive props for border radius

* updated responsive prop

* added responsive border radius story

* fixed box structure in story

* fixed console issue

* updated display constants

* updated constants

FrederikBolding committed:
[FLASK] Improve Snaps E2E stability (#16597)

* Improve Snaps E2E stability

* Simplify

* Fix lint

darkwing committed:
Create constants for all keyring types (#16575)

darkwing committed:
BETA - Update to sentence case (#16590)

darkwing committed:
Use async/await for extension functions (#15722)

bbehrman10 committed:
Make ENS named elements domain generic (#16166)

Co-authored-by: Olaf Tomalka <olaf.tomalka@gmail.com>
Co-authored-by: Vincent Shadbolt <vince.shadbolt@gmail.com>
Co-authored-by: Brad Decker <bhdecker84@gmail.com>
Co-authored-by: Brad Decker <git@braddecker.dev>

brad-decker commented:
@bbehrman10 can you rebase this for me today and by end of day i'll make a suggestion for the next slice to remove out for landing this in chunks. I know that seems like a longer process but ultimately its the only way we are going to get this landed in an auditable fashion.

lucasgday committed:
Expanded permission_ethereumAccounts spanish translation (#15660)

Co-authored-by: Brad Decker <bhdecker84@gmail.com>

bbehrman10 committed:
rebased

brad-decker commented:
Lets save anything to do with token resolution for the end, we don't currently do that at all and those changes to the send slice were the most alarming to me when reviewing this PR. The easiest to land would be the IPFS domain resolution, followed shortly thereafter by the recipient resolution that we already do. Once those two land you are at feature parity with what we currently support for ENS.

bbehrman10 commented:
sounds good, we will link to the new PR when it's ready for you to take a look at. Thanks Brad!

bbehrman10 commented:
@brad-decker we're trying to add the UD resolution package via yarn, but when running yarn dist we get a lavamoat error. We went through the process outlined in the github (setup, lavamoat:auto, etc), but still get the same package not in allowlist error. Are there new steps in order to add dependencies?

bbehrman10 commented:
Isn't that what the lavamoat commands are for? Rewriting those policy files so one doesn't have to manually put those keys in?

brad-decker commented:
There are two files "policy.json" and "policy-overrides.json" lavamoat:auto will build a policy file, so don't modify the 'policy.json' file. Not all dependency usages can be statically analyzed, and require a manual process. See https://github.com/LavaMoat/LavaMoat/blob/main/docs/policy.md#using-policy-overridejson-to-expand-or-limit-access

bbehrman10 commented:
@brad-decker is there a day or time that works well for you?

bbehrman10 commented:
IPFS Pull Request Ready for Review: https://github.com/MetaMask/metamask-extension/pull/16664

brad-decker commented:
@bbehrman10 could you push up your changes that are not passing lavamoat:auto ? its pretty hard to be helpful when looking at this PR because its so out of date.

V-Shadbolt committed:
updated tlds resolver keys package

V-Shadbolt committed:
updated tlds resolver keys package to latest
